great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -11,4 +15,10 @@ class AbstractDataContext(ABC):
     TODO: eventually the dependency on ConfigPeer will be removed and this will become a pure ABC.
     """
 
-    pass
+    @abstractmethod
+    def _init_variables(self) -> None:
+        raise NotImplementedError
```

Comment:
All DataContext classes are responsible for instantiating their specific variables instance per the contract set by the ABC.

Reply:
Ah makes sense, and then in the `__init__()` method you'll call `self._init_variables()`?

Reply:
yup! I'll have to work out exactly where this gets called (probably pair with Will) but that's the general idea